docs/xks/developer-guide/gitops.md

```diff
@@ -0,0 +1,71 @@
+---
+id: gitops
+title: GitOps a la XKS
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+## What is GitOps?
+
+> GitOps works by using Git as a single source of truth for declarative infrastructure and applications. With GitOps, the use of software agents can alert on any divergence between Git with what's running in [an environment], and if there's a difference, Kubernetes reconcilers automatically update or rollback the cluster depending on the case. &dash; _[Weave Works - Guide To GitOps](https://www.weave.works/technologies/gitops/)_
+
+XKS supports GitHub and Azure DevOps with almost identical workflows. XKF refers to as Git providers. For simplicity, we refer to their CI/CD automation as "pipelines". If you are using GitHub, whenever this text refers to "pipeline", think "GitHub Actions workflow". As you saw in the previous section, XKS comes with a set of pipelines that automatically detects app releases and promotes them through a series of environments. The allows both rapid iteration and strong validation of apps.
+
+XKS is built around [trunk-based development](https://trunkbaseddevelopment.com/).
+
+## Emilia updates an app
+
+In the previous section, we looked at deploying our first app to Kubernetes using a fully automatic flow. But what actually happened in that flow? This section tells the story about a developer called Emilia. She has updated an app packaged as a container image. A pipeline in the app's repository has just tagged the container image as `a8b91c33` and uploaded it to a container registry.
+
+The GitOps repository for Emilia's app has a `gitops-promotion.yaml` that looks like this (for much more details, see the [gitops-promotion](https://github.com/XenitAB/gitops-promotion) readme):
+
+```yaml
+prflow: per-app
+environments:
+  - env: dev
+    auto: true
+  - env: qa
+    auto: true
+  - env: prod
+    auto: false
+```
+
+The `dev` and `qa` environemnts have `auto: false` which means that new releases will be automatically applied, while `prod` environment is configured with `auto: false`. This means that pull requests must be merged by a human, presumably one that has verified that the update worked as expected in previous environments. The GitOps repository is configured to require checks on pull requests to pass in order to allow merge.
+
+### Applying to dev
+
+The flow is fully automatic and is triggered by the container image upload.
+
+<img alt="Apply to dev" src={useBaseUrl("img/assets/xks/developer-guide/developer-flow-apply-dev.jpg")} />
```

Comment:
The meaning of some of the symbols in the picture is not clear to me, especially the symbol what I believe denotes Flux. If you read the the text thoroughly you'll get it but I think it would be nice to understand it from just looking at the picture. Perhaps add text to the symbols or legend?

Reply:
My original notion was that this would mostly be done by a previous page which focuses on the various actors (kubelet, AKS, Flux, Terraform, et c). However, web documentation should always be standalone, so probably some sort of legend is in order. I'm going to take a stab at it.

Reply:
What do you think of this?
![image](https://user-images.githubusercontent.com/1666535/148641990-d2b5f7af-b15c-49e6-a4de-49bfa1a84479.png)